lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryWalker.php

```diff
@@ -66,7 +68,7 @@ public function walkSelectStatement(SelectStatement $AST)
         $pathExpression->type = PathExpression::TYPE_STATE_FIELD;
 
         $AST->selectClause->selectExpressions = [new SelectExpression($pathExpression, '_dctrn_id')];
-        $AST->selectClause->isDistinct        = true;
+        $AST->selectClause->isDistinct        = ! $query->hasHint(Paginator::HINT_DISABLE_DISTINCT) || ! $query->getHint(Paginator::HINT_DISABLE_DISTINCT);
```

Comment:
This would be less convoluted with something positively named, like `HINT_ENABLE_DISTINCT`. You wouldn't need to call `hasHint`, or to negate anything.

Reply:
I have tried this but `getHint` does not allow you to detect if the int is not there or is set to `false`. Since the `distinct` has to be used by default, i could not find a way to make it backward compat

Reply:
Only is to use `getHints` and then check if the hint is present in the array

Reply:
Oh right. So that would be the only hint defaulting to `true`? Or are there other cases?

Reply:
at the moment i see only true/false options, im not sure if it makes sense to have it as string and to call it "strategy" and have for now possible values such as: distinct (default) and none (new option).

Reply:
I'm OK with it being a boolean, I was more wondering about what other hints look like right now (and why there is this `false` default). If this hint you introduce is the only one that is defaulting to `true` then it might be best to revert to the previous situation with the negative naming like you did at the beginning.

Reply:
I've taken a look at other hints in the codebase, and it looks like this is the first time we have one that we want to default to `true`. So what you did is fine 👍